types/pipeline_events.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import "time"
+
+type PipelineEventSystem string
+
+const (
+	PipelineEventSystemDeployment PipelineEventSystem = "k8s:deployment"
+	PipelineEventSystemPod        PipelineEventSystem = "k8s:pod"
+)
+
+// PipelineEvent model.
+type PipelineEvent struct {
+	System     PipelineEventSystem `json:"system" yaml:"system"`
+	Status     PipelineStatusKind  `json:"kind" yaml:"kind"`
+	Reason     string              `json:"reason" yaml:"reason"`
+	Message    string              `json:"message" yaml:"message"`
+	LoggedAt   time.Time           `json:"loggedAt" yaml:"loggedAt"`
+}
+
+// CreatePipelineEvents is the format we submit the events in.
+type CreatePipelineEvent struct {
+	PipelineEvent
+}
+
+// CreatedPipelineEvent is the response we get when creating new pipeline
+// events.
+type CreatedPipelineEvent struct {
+	CreatePipelineEvent
+	StatusID  string    `json:"id" yaml:"id"`
```

Comment:
```suggestion
	OnStatusID  string    `json:"onStatusID" yaml:"onStatusID"`
```
I would rename it to something like this then. Otherwise it's confusing and makes you think its a new generated value.

Reply:
This makes me think there is a full eventing system here.. but meh.